### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by ALWAL12**
Any idea to improve the first draft of documentation?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,17 +1,93 @@
...
-**Range**: Can handle two comparables (overlapping for example). <br />
-**Pagination**: Structure that represents the pagination concept (page, itemPerPage, startingIndex). <br />
-**Picker**: Pick x item(s) randomly from a Collection. <br />
-**WeightedSet**: Pick a random item from a Collection. Each item appear x times according to their weight in the Set. <br />
+**DateProvider**
+```java
+// Useful for mocking purposes
+DateProvider dateProvider = new JavaDateProvider();
+
+// Without timezone
```
Comment (Vaub) [inline README.md]: I'd specify these as "legacy" and shouldn't be used

File: README.md [hunk 2]
```diff
@@ -1,17 +1,93 @@
...
+List<Integer> items = Arrays.asList(1, 2, 3, 4, 5);
+Optional<Integer> picked = new Picker<>(items).pickOne();
+List<Integer> threePicks = new Picker<>(items).pick(3);
+
+List<Integer> ignored = Arrays.asList(4, 5);
+List<Integer> thousandPickes = new Picker<>(items).pick(1000, ignored); // Will not contains any 4 or 5
+```
+
+**WeightedSet** <br />
+A set that apply a weight to each of its items. It is basically a lottery system. Each item has X chance(s) of being picked.
```
Comment (Vaub) [inline README.md]: [...] that applies [...] a lottery system where each item has [...]

File: README.md [hunk 3]
```diff
@@ -1,17 +1,93 @@
...
+Range<Integer> kids = Range.of(6, 12);
+Range<Integer> adults = Range.of(18, 65);
+
+boolean overlapping = kids.isOverlapping(adults); // false
+boolean overlapping = kids.isOverlapping(Range.of(0, 12)); // true
+```
+
+**Pagination**
+```java
+// Useful structure to represent pagination concept
```
Comment (Vaub) [inline README.md]: I'd put this under the title and not in the "code"

File: README.md [hunk 4]
```diff
@@ -1,17 +1,93 @@
...
+```
+
+**NumberPresenter**
+```java
+String display = new NumberPresenter(5.39332).numberOfDecimals(1).round().present();
+System.out.println(display); // 5.4
+```
+
+**ServiceLocator** <br />
+Super lightweight anti-pattern to inject dependencies easily. Useful for light testing.
```
Comment (Vaub) [inline README.md]: [...] to easily inject dependencies. [...] . 
Maybe put `anti-pattern` in quotation?

File: README.md [hunk 5]
```diff
@@ -1,17 +1,93 @@
...
+// Useful structure to represent pagination concept
+Pagination page = Pagination.paged(1, 10); // First ten items
+System.out.println(page.page()); // 1
+System.out.println(page.itemPerPage()); // 10
+System.out.println(page.startingIndex()); // 0
+
+Pagination offset = Pagination.offsetted(50, 10); // Skip the 50 first items, then take 10 items
+```
+
+**Picker**
```
Comment (Vaub) [inline README.md]: Maybe add a small description?

File: README.md [hunk 6]
```diff
@@ -1,17 +1,93 @@
 [![Build Status](https://travis-ci.org/Nexapp/NexappCore.svg?branch=master)](https://travis-ci.org/Nexapp/NexappCore)
 
 # Nexapp Core
-Several classes that we use everyday.
+Several Java mini-modules for REST APIs.
```
Comment (Vaub) [inline README.md]: Technically there's still nothing here for REST APIs

Comment (ALWAL12) [reply README.md]: Not about Jersey, but most classes are used in our APIs.

File: README.md (hunk 1 above)
Comment (ALWAL12) [reply README.md]: I'm unsure if this is really legacy. In fact, they are usable but not recommended. Java expose them as is, but we [Nexapp] prefer use the zoned version.

**Pull request merged**